### PR TITLE
Adding sinfo endpoint to slurmrestapi

### PR DESCRIPTION
**Pull request by alexxxxx**
Hello,

This pull request brings a new endpoint in the slurm rest api. This endpoint simply returns data in a similar format of sinfo.

3 files are touched:
rest/slurmrestapi.py which contains the endpoint definition
rest/utils.py which contains 2 methods to manipulate nodelist
tests/unit/test_utils.py which contains unit tests for rest/utils.py.

Unit test can be run using 
PYTHONPATH=$PYTHONPATH:. python tests/units/test_utils.py

Cheers,
Alex

### CODE REVIEW

**Comment by rezib** [comment]
Hi @alexxxxx,

First, thank you for your contribution!

I'm perfectly fine with the `/sinfo` route in `slurmrestapi` module. 

Concerning your code for `compact_nodes()` and `expand_nodes()` functions, we really prefer relying on `ClusterShell.NodeSet` library though. Check-out the documentation of this module for reference:
http://clustershell.readthedocs.io/en/latest/api/NodeSet.html

ClusterShell is already a dep of Slurm-web so we really recommend using it further.

File: rest/slurmrestapi.py [hunk 1]
```diff
@@ -425,6 +425,60 @@ def convert_nodeset():
...
+          'name': name,
+          'avail': avail,
+          'job_size': job_size,
+          'time_limit': time_limit,
+          'nodes': total_nodes,
+          'state': state,
+          'nodelist': compact_nodes(v)
+        })
+
+    # Jsonify can works on list, thus using json.dumps
```
Comment (rezib) [inline rest/slurmrestapi.py]: I guess you meant **_cannot_** work on list, right? Or I don't get the point. Please note Flask `jsonify()` not only concert a hash into a JSON string, it also sets HTTP headesr for properly handling JSON content.

**Comment by alexxxxx** [comment]
Thanks a lot with your quick feedback

+1 for cluster shell, I just tried to use it and it looks straightforward.

For return you're completely right, I meant cannot. The reason explained in [1] is that recursive structure can't be jsonify. Do you think it is fine is I use json.dumps and takes care of the headers?

http://flask.pocoo.org/docs/0.10/security/#json-security

**Comment by alexxxxx** [comment]
Now using Nodeset.

**Comment by rezib** [comment]
Thank you @alexxxxx!

**Pull request merged**